components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java

```diff
@@ -1737,6 +1737,154 @@ public static NativeObject jsFunction_getAllPaginatedPublishedAPIs(Context cx, S
+            throws ScriptException, APIManagementException {
+
+        APIConsumer apiConsumer = getAPIConsumer(thisObj);
+        String tenantDomain;
+        boolean returnAPItags = false;
+        boolean lightWeight = false;
+        String [] statusList = {APIConstants.PUBLISHED};
+        if (args[0] != null) {
+            tenantDomain = (String) args[0];
+        } else {
+            tenantDomain = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
+        }
+
+        int start = Integer.parseInt((String) args[1]);
+        int end = Integer.parseInt((String) args[2]);
+
+        if (args.length > 3 && args[3] != null) {
+            returnAPItags = Boolean.parseBoolean((String) args[3]);
+        }
+        return getPaginatedLightWeightAPIsByStatus(apiConsumer, tenantDomain, start, end, statusList, returnAPItags);
+    }
+
+    /**
+     * This method helps to get all paginated APIs by status.
+     * @param apiConsumer   ApiConsumer object to provide helper funcionality
+     * @param tenantDomain  Domain name of the tenant.
+     * @param start         The number to start the API Listing
+     * @param end           The number of end the API Listing
+     * @param status[]      Status of the API
+     * @param returnAPItags Optional parameter ,returns the tags of the API from registry if it is set to true
+     * @return this will return list of APIs by status.
+     */
+    private static NativeObject getPaginatedLightWeightAPIsByStatus(APIConsumer apiConsumer, String tenantDomain,
+                                                                    int start, int end, String[] status,
+                                                                    boolean returnAPItags) {
+        Set<API> apiSet;
+        Map<String, Object> resultMap;
+        NativeArray myn = new NativeArray(0);
+        NativeObject result = new NativeObject();
+
+        try {
+            if (tenantDomain != null && !MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
+                PrivilegedCarbonContext.startTenantFlow();
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain, true);
+            } else {
+                PrivilegedCarbonContext.startTenantFlow();
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain
+                        (MultitenantConstants.SUPER_TENANT_DOMAIN_NAME, true);
+            }
+            resultMap = apiConsumer.getAllPaginatedLightWeightAPIsByStatus(tenantDomain, start, end, status,
+                    returnAPItags);
+
+        } catch (APIManagementException e) {
+            log.error("Error from Registry API while getting API Information", e);
+            return result;
+        } catch (Exception e) {
+            log.error("Error while getting API Information", e);
+            return result;
+        } finally {
+            PrivilegedCarbonContext.endTenantFlow();
+        }
+        if (resultMap != null) {
```

Comment:
Can we move this block inside the try block? It will increase the readability of the code, WDYT?